File: BroadcastListener_2/Classes/Broadcaster.cs
using System.Collections.ObjectModel;
using System.Diagnostics;
using System.Windows.Forms;
using BroadcastListener_2.Interfaces;

namespace BroadcastListener_2.Classes
{
    public class Broadcaster
    {
        private readonly Collection<IMessageListener> _listeners = new Collection<IMessageListener>(); 

        /// <summary>
        /// Send message
        /// </summary>
        /// <param name="Message">Message</param>
        /// <remarks></remarks>
        [DebuggerStepThrough()]
        public void Broadcast(Person person, Form sender)
        {
            foreach (IMessageListener listener in _listeners)
            {
                listener.OnListen(person,sender);
            }
        }
       
        /// <summary>
        /// Add a Listener to the Collection of Listeners
        /// </summary>
        /// <param name="Listener"></param>
        public void AddListener(IMessageListener Listener)
        {
            _listeners.Add(Listener);
        }
        /// <summary>
        /// Remove a Listener from the collection
        /// </summary>
        /// <param name="Listener"></param>
        public void RemoveListener(IMessageListener Listener)
        {

            for (int index = 0; index < _listeners.Count; index++)
            {
                if ( _listeners[index].Equals(Listener) )
                {
                    _listeners.Remove(_listeners[index]);
                }
            }
        }
    }
}

File: ClassExample1/Classes/Broadcaster.vb
Imports System.Collections.ObjectModel

Public Class Broadcaster
    Private ReadOnly _listeners As New Collection(Of IMessageListener)()

    Public Sub Broadcast(person As Person, sender As Form)
        For Each listener As IMessageListener In _listeners
            listener.OnListen(person, sender)
        Next
    End Sub

    ''' <summary>
    ''' Add a Listener to the Collection of Listeners
    ''' </summary>
    ''' <param name="Listener"></param>
    Public Sub AddListener(Listener As IMessageListener)
        _listeners.Add(Listener)
    End Sub
    ''' <summary>
    ''' Remove a Listener from the collection
    ''' </summary>
    ''' <param name="Listener"></param>
    Public Sub RemoveListener(Listener As IMessageListener)

        Dim index As Integer = 0

        Do While index < _listeners.Count

            If _listeners(index).Equals(Listener) Then
                _listeners.Remove(_listeners(index))
            End If

            index += 1

        Loop
    End Sub
End Class
